### PR TITLE
Onboarding personalisation adjustments (retention#9)

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

- Adjusts the personalization screen based on latest developments from https://github.com/postHog/retention/issues/9.
- Conditionally asks the user if they have technical expertise based on their role response.
- Allows multiple selection and focusing selection on the `RadioSelect` component.

image

### CODE REVIEW

**Comment by Twixes** [comment]
A flyby note just from a glance at the gif: this UI seems SUPER wide. You may have heard about this nice book that includes some great tips on width, it's called "Refactoring UI". :D

File: frontend/src/scenes/onboarding/Personalization.tsx [hunk 1]
```diff
@@ -20,46 +19,37 @@ function _Personalization(): JSX.Element {
...
-    const handleOptionChanged = (attr: 'role' | 'team_size', value: string | null): void => {
+    const handleOptionChanged = (attr: 'role' | 'products' | 'technical', value: string | string[] | null): void => {
         appendPersonalizationData({ [attr]: value })
     }
 
     const handleContinue = (): void => {
         reportPersonalization(personalizationData, step, answeredQuestionCount === TOTAL_QUESTION_COUNT)
         // TODO: Update organization record
-        // :TODO: Is there a way to force default insights graph this without hard reload?
+        // TODO: Is there a way to force default insights graph this without hard reload?
```
Comment (macobo) [inline frontend/src/scenes/onboarding/Personalization.tsx:38]: Not a comment: :D Personal preference is to always use colon for these sort of comments.

File: frontend/src/lib/components/RadioSelect/index.tsx [hunk 2]
```diff
@@ -9,23 +11,75 @@ export interface RadioSelectType {
...
+            : selectedOption === option.key
+    }
+
+    const handleClick = (option: RadioSelectType | null): void => {
+        if (!option) {
+            onOptionChanged(null)
+            return
+        }
+
+        if (multipleSelection) {
```
Comment (macobo) [inline frontend/src/lib/components/RadioSelect/index.tsx:39]: :D This is quite messy - multi-selection with the same code tends to cause that.

But at the same time no direct business logic in here - good job!

File: frontend/src/lib/components/RadioSelect/index.tsx [hunk 3]
```diff
@@ -9,23 +11,75 @@ export interface RadioSelectType {
...
+            if (selectedOption instanceof Array) {
+                const _selectedOptions = selectedOption
```
Comment (macobo) [inline frontend/src/lib/components/RadioSelect/index.tsx:41]: This needs to be a copy - we should not mutate params like this.

`const _selectedOptions = [...selectedOption]`

**Review by macobo — APPROVED** [review]
Very minor suggestion below, otherwise lgtm.

File: posthog/api/team.py [hunk 4]
```diff
@@ -32,6 +32,30 @@ def has_permission(self, request: request.Request, view) -> bool:
         return True
 
 
+class TeamNestedSerializer(serializers.ModelSerializer):
```
Comment (macobo) [inline posthog/api/team.py:35]: Rebase fail?

File: frontend/src/lib/components/RadioSelect/index.tsx (hunk 2 above)
Comment (paolodamico) [reply frontend/src/lib/components/RadioSelect/index.tsx:39]: +1 agreed! thought about separating in two components but then the UI gets a bit messy, will keep it like this for now :)

File: posthog/api/team.py (hunk 4 above)
Comment (paolodamico) [reply posthog/api/team.py]: Seems this was an outdated view

**Pull request merged**